### DISCUSSION TITLE
query table with inner reference

### DISCUSSION THREAD

**Discussion by EdgarsLv** [supabase/supabase#9019] [Questions]
I have comments table where reply_to references id of same table
![image](https://user-images.githubusercontent.com/56837461/190878082-b5cbe856-5df1-46f9-9232-fab95aece3a7.png)

fetching like this i'm getting array with 6 object
![image](https://user-images.githubusercontent.com/56837461/190878117-15d9e41b-01da-4a34-a43a-b6617cb22d95.png)

but for this example i need to get array with 2 object for main comments where these 2 objects contains another array with  reply comment objects.

could look something like that:
`[{
discussion_id: 9,
id: 3,
reply_to: null,
text: "main comment one",
replies: [{...}, {...},{...}]
},
{...}
]`

It is possible to fetch like that? Or have to make another table just for replies and then somehow fetch?

**Comment by EdgarsLv** [comment]
so created another table for replies and now almost works as i expected when query like this:
![image](https://user-images.githubusercontent.com/56837461/190903439-a25ad7a3-abcc-407d-b5f9-23baaef9481c.png)
but now problem is that if main comment doesn't have replies then i'm not getting that comment at all and of course it will never get replies because no on will see it so won't be able to reply.

Really no one can help me out on this?

**Comment by GaryAustin1** [comment]
Try dropping the !inner part. Only used when you want to filter on that table.  
https://supabase.com/docs/reference/javascript/select#query-foreign-tables
